### DISCUSSION TITLE
Missing Thumbnails

### DISCUSSION THREAD

**Discussion by muddywellyboots** [biigle/core#379] [Q&A] [answered]
Hi,
I've installed the GPU branch of BIIGLE. I can't see any thumbnails, for images and annotations. Though images appear fine when they are opened. Any thoughts? I've looked through the logs for the various BIIGLE containers and can't see any errors.

Thanks
![Screenshot from 2021-09-22 22-44-21](https://user-images.githubusercontent.com/24653493/134426730-717582b5-0a4b-4b71-a158-8e8361202f13.png)

**Comment by mzur** [comment]
- Please have a look in the `public/storage/thumbs` directory. Is anything in there?
- Also, please run `docker-compose ps`. Are all containers running?
- Finally, have a look at `storage/logs/laravel.log`. Are there any error messages in the file?

**Comment by muddywellyboots** [comment]
Hi,
-Here is the contents of the `public/storage/thumbs`folder. Some of the folders contain second folders with thumbnails in.
![Screenshot from 2021-09-24 13-33-42](https://user-images.githubusercontent.com/24653493/134674895-4bd4fcb5-4ed4-4069-9dc6-236f8528f6e9.png)

-Yes, all containers seem to be running.
![Screenshot from 2021-09-24 13-34-57-dock](https://user-images.githubusercontent.com/24653493/134675055-f80c1929-c886-45c0-81c3-031d2ca5bae3.png)


I've attached a laravel.log from today (I uploaded an image and generated a report). Nothing jumps out as an error?

Thanks for your help
[laravel-2021-09-24.log](https://github.com/biigle/core/files/7225753/laravel-2021-09-24.log)

**Reply by mzur** [reply]
If you open the developer tools in the browser (<kbd>F12</kbd>), select the "Network" tab and then open or reload the volume overview (shown in your first screenshot), you should see requests with a `404` status in the network tab. These should attempt to load a file such as `17564613-7989-4de3-9133-ff1ad8ebfd2e.jpg`. With this file as example, do you find the corresponding directory `public/storage/thumbs/17/56` (directories are named after the beginning of the filename) which contains that file?

**Comment by muddywellyboots** [comment, marked as answer]
Good idea!
I tried it and there were no 404's. It looks like the image is coming from BIIGLE.

![Screenshot from 2021-09-24 16-25-53](https://user-images.githubusercontent.com/24653493/134701267-80253695-64c6-4fef-b84f-ba5264e9ca54.png)
o

**Reply by mzur** [reply]
Could you please show the content of the "network" tab as well?

**Reply by muddywellyboots** [reply]
I've reinstalled it, this time under a user 'ubuntu' and it works great. Problem solved. I wonder if it was a permissions thing...
Thanks for all your help.

![image](https://user-images.githubusercontent.com/24653493/134774306-3871231d-921d-4038-8dc4-8f9067318c0e.png)

**Comment by davidsean** [comment]
I had the same issue, and came here looking for answers. I found the fix by reading the above comments for tips.

In my case, the network tab showed a 404 for the url `https://example.com/storage/thumbs/93/12/9312d4e6-5fc4-419e-b38d-027f5f35f0a6.jpg`
The `example.com` part told me what was wrong. Turns out I did not populate `APP_URL` in `./build/.env`. Made the change, re-build the stack and it's now working.

**Comment by davidsean** [comment]
The issue came up again (after working for some time). New volumes/projects would not have thumbnails. There would exist a GET request for the thumbnail file, but no file was created at that path. No logs created. I am pretty sure all the needed docker containers were running.
I rebooted the host machine and after a while the thumbnails were created (as well as a log file for that day).

Here is what I think is the important part:
```
[2024-09-10 17:00:29] production.ERROR: Attempt to read property "id" on null (View: /var/www/resources/views/volumes/partials/projectsBreadcrumb.blade.php) (View: /var/www/resources/views/volumes/partials/projectsBreadcrumb.blade.php) {"userId":1,"exception":"[object] (Illuminate\\View\\ViewException(code: 0): Attempt to read property \"id\" on null (View: /var/www/resources/views/volumes/partials/projectsBreadcrumb.blade.php) (View: /var/www/resources/views/volumes/partials/projectsBreadcrumb.blade.php) at /var/www/storage/framework/views/c446b8ed8f0cc841620de06b2ac2c129.php:11)
```
Although it now works after a reboot, I am still posting in case it can help.
